lib/src/data/communication/api/api_errors.dart

```diff
@@ -0,0 +1,33 @@
+abstract class ApiError {
+  final int _statusCode;
+  final String _message;
+
+  ApiError(this._message, [this._statusCode]);
+
+  @override
+  String toString() => 'message: $_message,statusCode: $_statusCode';
+}
+
+class BadRequestError extends ApiError {
+  BadRequestError(statusCode, [message = "Bad request error"]) : super(message, statusCode);
+}
+
+class ServerError extends ApiError {
+  ServerError(statusCode, [message = "Server error"]) : super(message, statusCode);
+}
+
+class TimeoutError extends ApiError {
+  TimeoutError(statusCode, [message = "Timeout error"]) : super(message, statusCode);
```

Comment:
skoro wszędzie statusCode jest obligatoryjny to może w bazowej klasie również powinien?

Reply:
w przypadku błędu `ConnectionError` i `UnknownError` nie będzie statusCode